src/main/java/org/jitsi/videobridge/cc/AdaptiveTrackProjection.java

```diff
@@ -242,62 +235,66 @@ private static AdaptiveTrackProjectionContext makeContext(
     }
 
     /**
-     * Rewrites both RTP and RTCP packets and returns any additional packets
-     * that need to be piggy-backed.
+     * Rewrites an RTP packet and it returns any additional RTP packets that
+     * need to be piggy-backed.
      *
-     * @param rtpRtcpPacket an RTP or an RTCP packet to rewrite.
-     * @return any piggy-backed packets to include with the packet, or
-     * {@link #DROP_PACKET_ARR} if the packet needs to be dropped.
+     * @param rtpPacket the RTP packet to rewrite.
+     * @return any piggy-backed packets to include with the packet.
      */
-    public RawPacket[] rewrite(@NotNull RawPacket rtpRtcpPacket)
+    RawPacket[] rewriteRtp(@NotNull RawPacket rtpPacket)
+        throws
+        RewriteException
     {
-        if (rtpRtcpPacket.isInvalid())
-        {
-            return null;
-        }
-
         AdaptiveTrackProjectionContext contextCopy = context;
         if (contextCopy == null)
         {
             return EMPTY_PACKET_ARR;
         }
 
-        if (RTCPPacketPredicate.INSTANCE.test(rtpRtcpPacket))
+        RawPacketCache incomingRawPacketCache = null;
+        MediaStreamTrackDesc source = getSource();
+        if (source != null)
         {
-            return contextCopy.rewriteRtcp(rtpRtcpPacket)
-                ? EMPTY_PACKET_ARR
-                : DROP_PACKET_ARR;
-        }
-        else
-        {
-            RawPacketCache incomingRawPacketCache = null;
-            MediaStreamTrackDesc source = getSource();
-            if (source != null)
+            MediaStreamImpl stream
+                = source.getMediaStreamTrackReceiver().getStream();
+            if (stream != null)
             {
-                MediaStreamImpl stream
-                    = source.getMediaStreamTrackReceiver().getStream();
-                if (stream != null)
+                CachingTransformer cachingTransformer
+                    = stream.getCachingTransformer();
```

Comment:
some of these newlines look like they wouldn't go past 80 characters if they were on the same line?  or maybe it's just hard to tell on github

Reply:
Surprisingly they do exceed the 80 chars limit, but I've reformatted them in an effort to make them more readable.